File: server/datasets/java_source-code/Multiple-E2/type2_renamed_1_valid_parentheses.java
import java.util.HashMap;
import java.util.Map;
import java.util.Stack;

public class Solution {
    public boolean check_brackets(String text) {
        Stack<Character> memory_list = new Stack<>();
        Map<Character, Character> pairs = new HashMap<>();
        pairs.put(')', '(');
        pairs.put('}', '{');
        pairs.put(']', '[');
        
        for (char symbol : text.toCharArray()) {
            if (pairs.containsKey(symbol)) {
                char last_seen = memory_list.isEmpty() ? '#' : memory_list.pop();
                if (pairs.get(symbol) != last_seen) {
                    return false;
                }
            } else {
                memory_list.push(symbol);
            }
        }
        return memory_list.isEmpty();
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_e2/type2_renamed_1_valid_parentheses.py
def check_brackets(text):
    memory_list = []
    pairs = {")": "(", "}": "{", "]": "["}
    
    for symbol in text:
        if symbol in pairs:
            last_seen = memory_list.pop() if memory_list else '#'
            if pairs[symbol] != last_seen:
                return False
        else:
            memory_list.append(symbol)
            
    return not memory_list
